src/util/timelock.rs

```diff
@@ -0,0 +1,236 @@
+// Rust Bitcoin Library
+// Written in 2022 by
+//     Tobin C. Harding <me@tobin.cc>
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the CC0 Public Domain Dedication
+// along with this software.
+// If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+//
+
+//! Bitcoin timelocks.
+//!
+//! Relative and absolute timelocks based on `nSequence` and `nLockTime`.
+//!
+
+#[cfg(feature = "std")]
+use std::error;
+
+use core::fmt;
+
+use blockdata::constants::{
+    LOCKTIME_THRESHOLD, SEQUENCE_LOCKTIME_DISABLE_FLAG, SEQUENCE_LOCKTIME_GRANULARITY,
+    SEQUENCE_LOCKTIME_TYPE_FLAG,
+};
+
+/// An absolute time lock, expires after either height or time.
+///
+/// This is nLockTime as defined in BIP-65.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct Abs(u32);
```

Comment:
Maybe just me, but I immediately think "absolute value of an integer math function" (like https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/abs).

I think `Absolute` would have less of this problem.

Reply:
That is true, when qualified with a module it does seem OK (though will people use it like that always - hard to say). Also no strong feelings, more of a subjective perspective.

Reply:
I used the locks exclusively as `timelock::Abs`/`timelock::Rel` I believe, as Andrew mentions, but now, a day or two later, looking at `Abs` it does seem the name is too terse. I'll have a play around with your suggestion @dpc.

Reply:
Went to do this change and soon as I saw 'Absolute' I was like "absolute what?" so I think this does no fix the problem. The problem is 'absolute' is an overloaded term, 'absolute value' and 'absolute timelock'.

Reply:
`AbsTimelock`? `AbsoluteTimelock`?

Or should `Timelock` be just `enum Timelock { Absolute { ... }, Relative { ... }}` so qualified name use is more common?

Just thinking aloud.

Reply:
I think we should leave it as is (I don't mind having longer names), and users can use `timelock::Absolute`.

Adding our own `Timelock` suffix saves, at best, two characters, at the expense of making it less clear where these types come from.

Reply:
FTR, I'm not fussed so will go with the long versions `timelock::Absolute` and `timelock::Relative`.